### ISSUE TITLE
Make forest gradually paler on lower zoom levels

### ISSUE THREAD

**Issue by matkoniecz** [gravitystorm/openstreetmap-carto#1387]
Currently there are two problems. Forest are dominating everything else on z8, z9. Then forests completely disappear on z7.

Google have much more readable and prettier map on lower zoom levels, in part thanks to displaying paler forests across broader range of zoom levels.

Making paler forests across all zoom levels would not work as we are trying to display bushes, grass in distinct styles at higher zoom levels etc 

But making forests gradually paler has chance to work and allow us to display forests at z7 and increase readability of z8 and z9.

This idea is result of using http://tools.geofabrik.de/mc/#7/51.8881/10.5610&num=2&mt0=mapnik&mt1=google-map

**Comment by imagico** [comment]
Landcover rendering at low zooms is a complex matter and is IMO not going to be solved just by tweaking colors and extending rendering to lower zooms at the cost of performance.  That being said the [German style](http://www.openstreetmap.de/karte.html?zoom=7&lat=51.32747&lon=10.73945&layers=B000TF) modifies the forest color at low zoom to be less disturbing.  This however also makes it more difficult to discern different types of landcovers.

**Comment by daganzdaanda** [comment]
I like that idea! 
On z=7, we could have a very faint green for the forests, and then increase the saturation and darken it in a few steps so that at z=12 we have the "final" colour. At z=12, we could start showing symbol patterns, as discussed in #1242. 
IMHO a  review of the style of forests should be high on our to-do-list, since they are still left over from the landcover desaturation. And after forests are adjusted, we can go work to make roads more visible (#102)

**Comment by matkoniecz** [comment]
I will try to tweak at least z8. Without that it will be quite hard to have decent rendering both in roads going through heavily forested areas and in region with high road density (making roads more visible improves situation in the first case and decreases in the second one).

Also  - currently there is a massive chance how map looks on zooming in to z8 from z7 when forest appears. 

Slovakia - https://cloud.githubusercontent.com/assets/899988/8614400/900d7af0-26e9-11e5-93b5-4b62f22cc3a5.png

**Comment by matkoniecz** [comment]
And area with high road density (sorry, only current version of proposed road style, location is http://www.openstreetmap.org/#map=8/51.1678/7.0043 ). It shows that "make road more noticeable" is going to move problem into other locations. Paler forests on z8, maybe also z9 seems to be a better idea.

Germany - https://cloud.githubusercontent.com/assets/899988/8627506/d7e2f046-274b-11e5-9264-9d9501818796.png

**Comment by math1985** [comment]
> It shows that "make road more noticeable" is going to move problem into other locations

Sorry, what do you mean exactly by this?

**Comment by matkoniecz** [comment]
The problem with road in forest on z8/z9 may be solved by making road wider or changing colour into more noticeable.

But it would cause areas with high road density to become even more cluttered, some places like London or west Germany would be in danger of making it impossible to distinguish separate roads.

See for example

![selection_001](https://cloud.githubusercontent.com/assets/899988/8629690/e764f9e4-275d-11e5-930e-b5dc5588edd7.png)

at the same time so regions have roads being not noticeable enough and roads with too prominent rendering (red motorway/trunk are not afffected, problem is with yellow/orange secondary/tertiary).

Making forests paler reduce the "not visible in forest" problem.

**Comment by math1985** [comment]
@matkoniecz Is this issue still relevant?

**Comment by pnorman** [comment]
> @matkoniecz Is this issue still relevant?

It's still an issue, see also #1755 which is about contrast between forest and residential, comments on https://github.com/gravitystorm/openstreetmap-carto/pull/1736#issuecomment-131495833 about forest/road contrast

**Comment by nebulon42** [comment]
Ah, forgot that this issue existed. I wanted to experiment with it, but testing is difficult without larger areas and unfortunately my database resources are limited. Currently I have only Austria in my database and that is already 8GB. So the only thing I can do is to say that would very much support fading in forests.

**Comment by pnorman** [comment]
> Ah, forgot that this issue existed. I wanted to experiment with it, but testing is difficult without larger areas and unfortunately my database resources are limited. Currently I have only Austria in my database and that is already 8GB. So the only thing I can do is to say that I would very much support fading in forests

Try a lower data density area. British Columbia has a **lot** of forests with not too many dense areas. Japan has some areas, but doesn't come as a handy extract.

**Comment by nebulon42** [comment]
> Try a lower data density area.

Worth a try. Thanks for the hint.

**Comment by nebulon42** [comment]
Here is a suggestion, this is by no means finished, but it is a start. Open questions are whether the range is too wide and how to distribute the changes on the L and the c value.

By the way, does anybody know a decent Lch colour picker? The one I found accepted only h=136, but I needed h=135.

What also becomes apparent, especially on z10, is that adjusting forest is not enough, but also other landuse colours would need adjustment.

If you want to play with the code, the branch is here: https://github.com/nebulon42/openstreetmap-carto/tree/forest-low-zoom

z7 (no forest)
![forest_z7](https://cloud.githubusercontent.com/assets/3531092/9342809/426d78ca-45fe-11e5-88f8-540a922e964e.png)

z8
![forest_z8](https://cloud.githubusercontent.com/assets/3531092/9342812/4b75a2e4-45fe-11e5-8ecc-795021a2f2a6.png)

z9
![forest_z9](https://cloud.githubusercontent.com/assets/3531092/9342815/50812f74-45fe-11e5-85a1-ea7b72192e15.png)

z10
![forest_z10](https://cloud.githubusercontent.com/assets/3531092/9342820/5447997c-45fe-11e5-9c38-a6cbfb56493d.png)

z11
![forest_z11](https://cloud.githubusercontent.com/assets/3531092/9342823/576f7778-45fe-11e5-969c-f8c6044a27c7.png)

z12 (original colour reached)
![forest_z12](https://cloud.githubusercontent.com/assets/3531092/9342830/6784b448-45fe-11e5-913b-85fb8750a193.png)

I can also show a preview for British Columbia, but I picked Austria because it has more diverse features.

**Comment by daganzdaanda** [comment]
Looks good already! 
Which other landuses would you want to adjust at z10? All of them?
Is there some farmland in that area at all?
Is it possible to show this with the new road colours already?

**Comment by matkoniecz** [comment]
I thoought about first zoom level for forest at z7 or even z6 (earlier than now), with max reached at z10 (to avoid rebalancing huge amount of landuses).

From examples - it looks really promising.

**Comment by pnorman** [comment]
> By the way, does anybody know a decent Lch colour picker? The one I found accepted only h=136, but I needed h=135.

I use two tools. For simple rgb/lch conversions, I use http://colormine.org/convert/rgb-to-lch. For picking colours, I use the "full-blown example showing all the sliders in action" example on http://www.virtuosoft.eu/code/jquery-colorpickersliders/. This picker is good because it lets you see the cutoffs. For example, if you have Lightness of 100, you can't have any chroma, and there aren't any high-chroma dark yellows.

**Comment by pnorman** [comment]
I don't think it works particularly well when overlapping with nature reserves

z8
![image](https://cloud.githubusercontent.com/assets/1190866/9345018/e1579fa2-45c1-11e5-8142-227cbc991ebf.png)

**Comment by nebulon42** [comment]
Thanks for the comments. I think at least meadows/grassland would need adjustment. I will try to show some previews including farmland and new road colours too.

I wanted to have z11 paler too, because there the new road style still has problems, which you tried to solve with the glow effect. As you can't show casings at z11, forest should probably be less visually heavy there. Adjusting other landuses opens up a lot of new problems, but I think it might be needed.

**Comment by matkoniecz** [comment]
> I don't think it works particularly well when overlapping with nature reserves

I have no problems with that situation - and for me it is clearly better than current situation. 

Maybe making also nature reserve paler would improve situation.

**Comment by imagico** [comment]
I think the new road colors need to be considered here.

In general i have my doubt that tuning individual colors at the lower zooms will do much good.  You could try fading out all landcover colors in a common way but ultimately the main problem with the low zooms is that the AGG renderer is simply incapable of properly rendering landcover with geometries that are very detailed and finely split compared to the rendering resolution.  In most well mapped areas this applies to ~z<=9.

**Comment by nebulon42** [comment]
Agreed, new road colours are important here. I'll try to take this further the next days or at the weekend. I see it as proof of concept or something similar, which means interesting to experiment with, but might be never merged. Let's see where we get.

**Comment by mboeringa** [comment]
> Looks good already!

\+1

> I have no problems with that situation - and for me it is clearly better than current situation.

\+1, I think it looks pretty acceptable, the outline and label within it clearly signify the boundaries. Maybe the boundaries of the national_park / protected_area / nature_reserves could be made thinner at low zoom (Z7/8), and then broadened zooming in.

**Comment by Rovastar** [comment]
Not sure if I like this. Consistency is a good thing. How on earth can you have a sensible legend/key with these constantly changing colours? I thought we were trying to consolidate the dozen or so greens not make another 5 more.

**Comment by mboeringa** [comment]
> Not sure if I like this. Consistency is a good thing. How on earth can you have a sensible legend/key with these constantly changing colours? I thought we were trying to consolidate the dozen or so greens not make another 5 more.

That's a valid good point, maybe it would be better to only subdue the colors in one step, instead of gradually, and only for those zoomlevels where forests are the only landuse feature being rendered, which seams just Z8?

**Comment by matkoniecz** [comment]
> How on earth can you have a sensible legend/key with these constantly changing colours?

Maybe legend changing on zooming in/out? It already displays/hides entries depending on current zoom level.

**Comment by matkoniecz** [comment]
@nebulon42 Thanks for your work! Within last days I experimented with it, and it was a very useful base for tweaking things.

@pnorman 
> I don't think it works particularly well when overlapping with nature reserves

I agree, especially as at this moment protected areas on lower zoom levels are rendered in way that makes them looks like forests. So I experimented with changing rendering to nature reserves, including removal of fill and tweaking borders and some other changes.

For example #2199 (do not display borders between national parks on lower zoom levels) is part of that.

Images below are displaying one of variants and include #2199 proof of concept with all its current flaws.

https://cloud.githubusercontent.com/assets/899988/16359372/48217472-3b32-11e6-992f-5afea9913688.png
https://cloud.githubusercontent.com/assets/899988/16359373/48221cc4-3b32-11e6-8a41-f3f13231bb1d.png
https://cloud.githubusercontent.com/assets/899988/16359374/48227322-3b32-11e6-835a-119d5fea9190.png
https://cloud.githubusercontent.com/assets/899988/16359371/48212602-3b32-11e6-8436-2ddbd774e683.png
